SecretManagement.LastPass.psm1

```diff
@@ -0,0 +1,114 @@
+﻿# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+using namespace Microsoft.PowerShell.SecretManagement
+
+$ModuleName = 'SecretManagement.LastPass'
+
+# The last segement (underscore + number (eg: _1)) is to view how many format args are expected.
+$ErrorMessages = @{
+    GetVaultParams0                 = "At least 1 vault implementing $ModuleName must be registered."
+    GetVaultParamsMany_1            = "`$Vault argument must be provided when multiple vault implementing $ModuleName exists: {0}"
+    StayConnectedForceSwitchMissing = 'StayConnected is a sensitive operation that save the LastPass decryption key on your hard drive. If you want to proceed, reissue the command specifying the force parameter.'
+    Unregister_NotLpass_1           = "The specified vault is not a $ModuleName vault (VaultType: {0}"
+}
+
+function Connect-LastPass {
+    [CmdletBinding()]
+    param (
+        [String]$Vault,
+        [String]$User,
+        [Switch]$Trust,
+        [Switch]$StayConnected,
+        [Switch]$Force 
+    )
+    $Arguments = [System.Collections.Generic.List[String]]@('login')
+    if ($trust) { $Arguments.Add('--trust') }
+    if ($StayConnected) { 
+        if (! $Force) { Throw [System.Management.Automation.PSArgumentException] $ErrorMessages.StayConnectedForceSwitchMissing }
```

Comment:
This should be implemented using ShouldProcess mechanics:

https://docs.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-shouldprocess?view=powershell-7.1#implementing--force

Reply:
https://github.com/TylerLeonhardt/SecretManagement.LastPass/pull/18/commits/2fc9336324651b26c959788b9c8cdd0b4f497d09